### PR TITLE
gtkmm4: skip check on darwin

### PR DESCRIPTION
**Pull request by 2gn**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
Objection. Builds on darwin with `doCheck = stdenv.isLinux;`.

**Comment by 2gn** [comment]
> Objection. Builds on darwin with `doCheck = stdenv.isLinux;`:
> 
> ```
> ...
> [380/383] Compiling C++ object demos/gtk-demo/gtkmm-demo.p/example_treeview_editable_cells.cc.o
> [381/383] Compiling C++ object demos/gtk-demo/gtkmm-demo.p/example_treeview_liststore.cc.o
> [382/383] Compiling C++ object demos/gtk-demo/gtkmm-demo.p/example_treeview_treestore.cc.o
> [383/383] Linking target demos/gtk-demo/gtkmm-demo
> buildPhase completed in 1 minutes 36 seconds
> ```

```
Installing gtk/gtkmm/libgtkmm-4.0.0.dylib to /nix/store/a5j0h237n4h6svkp8w3cygxv3bd3qx3i-gtkmm-4.6.1/lib
Installing symlink pointing to libgtkmm-4.0.0.dylib to /nix/store/a5j0h237n4h6svkp8w3cygxv3bd3qx3i-gtkmm-4.6.1/lib/libgtkmm-4.0.dylib
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/gdk/gdkmm/../gdkmm.h to /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/gdk/gdkmm/wrap_init.h to /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0/gdkmm
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/gdk/gdkmm/cairoutils.h to /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0/gdkmm
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/gdk/gdkmm/devicewithpad.h to /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0/gdkmm
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/gdk/gdkmm/dragsurfaceimpl.h to /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0/gdkmm
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/gdk/gdkmm/general.h to /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0/gdkmm
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/gdk/gdkmm/popupsurfaceimpl.h to /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0/gdkmm
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/gdk/gdkmm/toplevelsurfaceimpl.h to /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0/gdkmm
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/gdk/gdkmm/value_cairo.h to /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0/gdkmm
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/gtk/gtkmm/../gtkmm.h to /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/gtk/gtkmm/version.h to /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0/gtkmm
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/gtk/gtkmm/wrap_init.h to /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0/gtkmm
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/gtk/gtkmm/accelerator.h to /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0/gtkmm
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/gtk/gtkmm/accelkey.h to /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0/gtkmm
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/gtk/gtkmm/bitsetconstiter.h to /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0/gtkmm
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/gtk/gtkmm/cellrenderer_generation.h to /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0/gtkmm
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/gtk/gtkmm/init.h to /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0/gtkmm
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/gtk/gtkmm/listviewtext.h to /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0/gtkmm
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/gtk/gtkmm/object.h to /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0/gtkmm
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/gtk/gtkmm/selectionlistmodelimpl.h to /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0/gtkmm
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/gtk/gtkmm/treemodelcolumn.h to /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0/gtkmm
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/gtk/gtkmm/treeview_private.h to /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0/gtkmm
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/gtk/gtkmm/private/object_p.h to /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0/gtkmm/private
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/tools/extra_defs_gen/../m4/class_gtkobject.m4 to /nix/store/a5j0h237n4h6svkp8w3cygxv3bd3qx3i-gtkmm-4.6.1/lib/gtkmm-4.0/proc/m4
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/tools/extra_defs_gen/../m4/convert.m4 to /nix/store/a5j0h237n4h6svkp8w3cygxv3bd3qx3i-gtkmm-4.6.1/lib/gtkmm-4.0/proc/m4
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/tools/extra_defs_gen/../m4/convert_gdk.m4 to /nix/store/a5j0h237n4h6svkp8w3cygxv3bd3qx3i-gtkmm-4.6.1/lib/gtkmm-4.0/proc/m4
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/tools/extra_defs_gen/../m4/convert_gtk.m4 to /nix/store/a5j0h237n4h6svkp8w3cygxv3bd3qx3i-gtkmm-4.6.1/lib/gtkmm-4.0/proc/m4
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/tools/extra_defs_gen/../m4/convert_gtkmm.m4 to /nix/store/a5j0h237n4h6svkp8w3cygxv3bd3qx3i-gtkmm-4.6.1/lib/gtkmm-4.0/proc/m4
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/build/gdk/gdkmmconfig.h to /nix/store/a5j0h237n4h6svkp8w3cygxv3bd3qx3i-gtkmm-4.6.1/lib/gtkmm-4.0/include
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/build/gtk/gtkmm-4.0.pc to /nix/store/a5j0h237n4h6svkp8w3cygxv3bd3qx3i-gtkmm-4.6.1/lib/pkgconfig
Installing /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/build/gtk/gtkmmconfig.h to /nix/store/a5j0h237n4h6svkp8w3cygxv3bd3qx3i-gtkmm-4.6.1/lib/gtkmm-4.0/include
Running custom install script '/nix/store/q39clx0r9fy4aki89kx8ylxlnvaylvhj-python3-3.10.5/bin/python3.10 /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/untracked/build_scripts/generate-binding.py install_built_h_files /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/untracked/gdk/gdkmm /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0/gdkmm applaunchcontext cairocontext clipboard contentformats contentformatsbuilder contentprovider cursor device devicepad devicetool display displaymanager drag dragsurface drawcontext drop enums event frameclock frametimings glcontext gltexture memorytexture monitor paintable pixbuf pixbufanimation pixbufanimationiter pixbufformat pixbufloader popup popuplayout rectangle rgba seat snapshot surface texture timecoord toplevel toplevellayout toplevelsize'
Running custom install script '/nix/store/q39clx0r9fy4aki89kx8ylxlnvaylvhj-python3-3.10.5/bin/python3.10 /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/untracked/build_scripts/generate-binding.py install_built_h_files /private/tmp/nix-build-gtkmm-4.6.1.drv-0/gtkmm-4.6.1/untracked/gtk/gtkmm /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/include/gtkmm-4.0/gtkmm aboutdialog accessible actionable actionbar adjustment appchooser appchooserbutton appchooserdialog appchooserwidget application applicationwindow aspectframe assistant assistantpage binlayout bitset boolfilter border box boxlayout buildable builder buildercscope builderscope button calendar cellarea cellareabox cellareacontext celleditable celllayout cellrenderer cellrendereraccel cellrenderercombo cellrendererpixbuf cellrendererprogress cellrendererspin cellrendererspinner cellrenderertext cellrenderertoggle cellview centerbox centerlayout checkbutton colorbutton colorchooser colorchooserdialog columnview columnviewcolumn combobox comboboxtext constraint constraintguide constraintlayout constrainttarget csslocation cssprovider csssection dialog directorylist dragicon dragsource drawingarea dropcontrollermotion dropdown droptarget droptargetasync editable editablelabel emojichooser entry entrycompletion entrybuffer enums eventcontroller eventcontrollerfocus eventcontrollerkey eventcontrollerlegacy eventcontrollermotion eventcontrollerscroll expander expression expressionwatch filechooser filechooserdialog filechoosernative filechooserwidget filefilter filter filterlistmodel fixed flattenlistmodel flowbox flowboxchild fontbutton fontchooser fontchooserdialog fontchooserwidget frame gesture gestureclick gesturedrag gesturelongpress gesturepan gesturerotate gesturesingle gesturestylus gestureswipe gesturezoom glarea grid gridlayout gridlayoutchild gridview headerbar iconpaintable icontheme iconview image infobar label layoutchild layoutmanager levelbar linkbutton listbase listbox listboxrow listitem listitemfactory liststore listview lockbutton mediacontrols mediafile mediastream menubutton messagedialog multifilter multiselection multisorter native nativedialog noselection notebook notebookpage numericsorter orientable overlay padactionentry padcontroller pagesetup paned papersize passwordentry picture popover popovermenu popovermenubar printcontext printoperation printoperationpreview printsettings progressbar range recentinfo recentmanager requisition revealer root scale scalebutton scrollable scrollbar scrolledwindow searchbar searchentry selectionfiltermodel selectionmodel separator settings signallistitemfactory shortcut shortcutaction shortcutcontroller shortcutlabel shortcutmanager shortcutsgroup shortcutssection shortcutsshortcut shortcutswindow shortcuttrigger singleselection sizegroup slicelistmodel snapshot sorter sortlistmodel spinbutton spinner stack stackpage stacksidebar stackswitcher statusbar stringfilter stringlist stringobject stringsorter stylecontext styleprovider switch text textbuffer textchildanchor textiter textmark texttag texttagtable textview togglebutton tooltip treedragdest treedragsource treeexpander treeiter treelistmodel treelistrow treemodel treemodelfilter treemodelsort treepath treerowreference treeselection treesortable treestore treeview treeviewcolumn video viewport volumebutton widget widgetpaintable window windowcontrols windowgroup windowhandle pagesetupunixdialog printer printjob printunixdialog'
glibPreFixupPhase
dropIconThemeCache
post-installation fixup
Moving /nix/store/a5j0h237n4h6svkp8w3cygxv3bd3qx3i-gtkmm-4.6.1/lib/pkgconfig to /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/lib/pkgconfig
rmdir: failed to remove '/nix/store/a5j0h237n4h6svkp8w3cygxv3bd3qx3i-gtkmm-4.6.1/lib': Directory not empty
Patching '/nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/lib/pkgconfig/gtkmm-4.0.pc' includedir to output /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev
strip is /nix/store/yr4ifz2ryq1c12jcqg6n1rkrwbzw8ml1-cctools-binutils-darwin-949.0.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/a5j0h237n4h6svkp8w3cygxv3bd3qx3i-gtkmm-4.6.1/lib 
patching script interpreter paths in /nix/store/a5j0h237n4h6svkp8w3cygxv3bd3qx3i-gtkmm-4.6.1
strip is /nix/store/yr4ifz2ryq1c12jcqg6n1rkrwbzw8ml1-cctools-binutils-darwin-949.0.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev/lib 
patching script interpreter paths in /nix/store/6ya9wm5pigqwlki40g72wafiln6jw06a-gtkmm-4.6.1-dev
/nix/store/a5j0h237n4h6svkp8w3cygxv3bd3qx3i-gtkmm-4.6.1
```

It seems to be working on my environment as well. Thank you.

**Comment by jiegec** [comment]
Result of `nixpkgs-review pr 183983` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gtkmm4</li>
  </ul>
</details>

**Comment by jtojnar** [comment]
What was the error? Is there no easy way to fix it?

**Comment by wegank** [comment]
The tests run well on my machine. Not sure if it is the case on Hydra, though.

```nix
  checkInputs = lib.optionals (!stdenv.isDarwin) [
    xvfb-run
  ];

  checkPhase = ''
    runHook preCheck

    ${lib.optionalString (!stdenv.isDarwin) "xvfb-run -s '-screen 0 800x600x24' \\"}
      meson test --print-errorlogs

    runHook postCheck
  '';
```

**Comment by 2gn** [comment]
> What was the error? Is there no easy way to fix it?

It fails to build package `xvfb-run`, which seems to work only on X environment.

**Comment by wegank** [comment]
> > What was the error? Is there no easy way to fix it?
> 
> It fails to build package `xvfb-run`, which seems to work only on X environment.

We don't need xvfb-run for the test to pass on Darwin. Could you try my fix in my comment above to see if it builds on Hydra?

**Review by wegank — APPROVED** [review]

File: pkgs/development/libraries/ncine/default.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  makeFlags = [
+    "SHELL=${stdenv.shell}"
+    "INSTALL_ROOT=$(out)"
+  ] ++ optional stdenv.cc.isClang "CC=clang";
+
+  preInstall = "mkdir -p $out/{bin,lib,man/man1,man/man3,include/gsm}";
+
+  parallelBuild = false;
+
+  meta = with lib; {
```
Comment (jiegec) [inline pkgs/development/libraries/ncine/default.nix:43]: This seems unrelated to the pr?

Comment (wegank) [reply pkgs/development/libraries/ncine/default.nix:43]: Oh I didn't see that.

**Comment by 2gn** [comment]
Sorry. I noticed that I've mistakenly committed changes on the package `nCine`, which is totally not related to this pr. I'm going to fix this.

File: pkgs/development/libraries/ncine/default.nix (hunk 1 above)
Comment (2gn) [reply pkgs/development/libraries/ncine/default.nix:43]: I just noticed this. Maybe I've mistakenly executed `git add .` without changing the branch or something. Thank you.

**Comment by jiegec** [comment]
Please squash the unwanted commit.

**Comment by jtojnar** [comment]
Thanks.
